charts/airbyte/templates/worker/deployment.yaml

```diff
@@ -0,0 +1,241 @@
+        - name: RESOURCE_MEMORY_REQUEST
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: RESOURCE_MEMORY_REQUEST
+        - name: RESOURCE_MEMORY_LIMIT
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: RESOURCE_MEMORY_LIMIT
+        - name: S3_LOG_BUCKET
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: S3_LOG_BUCKET
+        - name: S3_LOG_BUCKET_REGION
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: S3_LOG_BUCKET_REGION
+        - name: AWS_ACCESS_KEY_ID
+          value: {{ .Values.minio.accessKey.password }}
+        - name: AWS_SECRET_ACCESS_KEY
+          value: {{ .Values.minio.secretKey.password }}
+        - name: S3_MINIO_ENDPOINT
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: S3_MINIO_ENDPOINT
+        - name: S3_PATH_STYLE_ACCESS
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: S3_PATH_STYLE_ACCESS
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: GOOGLE_APPLICATION_CREDENTIALS
+        - name: GCP_STORAGE_BUCKET
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: GCP_STORAGE_BUCKET
+        - name: INTERNAL_API_HOST
+          valueFrom:
+            configMapKeyRef:
+              name: airbyte-env
+              key: INTERNAL_API_HOST
+        {{- if .Values.worker.extraEnv }}
+        {{ .Values.worker.extraEnv | toYaml | nindent 8 }}
+        {{- end }}
+        {{- if .Values.worker.livenessProbe.enabled }}
+        livenessProbe:
+          httpGet:
+            path: /
+            port: heartbeat
+          initialDelaySeconds: {{ .Values.worker.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.worker.livenessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.worker.livenessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.worker.livenessProbe.successThreshold }}
+          failureThreshold: {{ .Values.worker.livenessProbe.failureThreshold }}
+        {{- end }}
+        {{- if .Values.worker.readinessProbe.enabled }}
+        readinessProbe:
+          httpGet:
+            path: /
+            port: heartbeat
+          initialDelaySeconds: {{ .Values.worker.readinessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.worker.readinessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.worker.readinessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.worker.readinessProbe.successThreshold }}
+          failureThreshold: {{ .Values.worker.readinessProbe.failureThreshold }}
+        {{- end }}
+        ports:
+        - name: heartbeat
```

Comment:
Gave this port a name for using with readiness/liveness probes. If its not valid to use this port for that, I can remove the probes and the name given to this port. From looking at https://github.com/airbytehq/airbyte/blob/ef853153a581dbb0a94c8aec3a5651cf6fa4de51/airbyte-workers/src/main/java/io/airbyte/workers/process/WorkerHeartbeatServer.java#L90-L98 it looks like this should work.

Reply:
works well!